src/readonly-table/index.ts

```diff
@@ -1,220 +1,213 @@
+            ${this.label}
+          </caption>
+          <thead class="kuc-readonly-table__table__header">
+            <tr>
+              ${this.columns.map((column) => this._getColumnsTemplate(column))}
+            </tr>
+          </thead>
+          <tbody class="kuc-readonly-table__table__body">
+            ${currentPageData.map((data: object, currentIndex: number) => {
+              return this._getDataTemplate(
+                data,
+                currentIndex,
+                this._columnOrder
+              );
+            })}
+          </tbody>
+        </table>
+        <kuc-base-pagination
+          .visible="${this.pagination}"
+          .isPrev="${this._toggleDisplayPreviousButton()}"
+          .isNext="${this._toggleDisplayNextButton()}"
+          @kuc:pagination-click-prev=${this._handleClickPreviousButton}
+          @kuc:pagination-click-next=${this._handleClickNextButton}
+        ></kuc-base-pagination>
+      `;
+    }
 
-  update(changedProperties: PropertyValues) {
-    if (changedProperties.has("columns")) this._validateColumns(this.columns);
-    if (changedProperties.has("data")) this._validateData(this.data);
+    private _createDisplayData(
+      data: object[],
+      pagePosition: number,
+      steps: number
+    ) {
+      if (!this.pagination) return data;
+      const firstRow = (pagePosition - 1) * steps + 1;
+      const lastRow = pagePosition * steps;
+      const displayData = data.filter(
+        (_element, index: number) =>
+          index >= firstRow - 1 && index <= lastRow - 1
+      );
+      return displayData;
+    }
 
-    super.update(changedProperties);
-  }
+    private _getColumnsTemplate(column: Column) {
+      return html`
+        <th
+          class="kuc-readonly-table__table__header__cell"
+          ?hidden="${column.visible === false}"
+        >
+          <span class="kuc-readonly-table__table__header__cell__label"
+            >${column.title}</span
+          >
+        </th>
+      `;
+    }
 
-  private _getColumnsTemplate(column: Column) {
-    return html`
-      <th
-        class="kuc-readonly-table__table__header__cell"
-        ?hidden="${column.visible === false}"
-      >
-        <span class="kuc-readonly-table__table__header__cell__label">
-          ${column.header && column.header.text}</span
+    // Formatting the data displayed on the current page
+    private _getDataTemplate(
+      data: any,
+      currentIndex: number,
+      columnOrder: string[]
+    ) {
+      return html`
+        <tr
+          class="kuc-readonly-table__table__body__row kuc-readonly-table__table__body__row-${currentIndex}"
         >
-      </th>
-    `;
-  }
+          ${columnOrder.map((currentCol, colIndex) => {
+            let isHidden = false;
+            if (
+              this.columns[colIndex] &&
+              this.columns[colIndex].visible === false
+            ) {
+              isHidden = true;
+            }
+            /* eslint-disable */
```

Comment:
what is this?

Reply:
return html`
              <td
                class="kuc-readonly-table__table__body__row__cell-data"
                ?hidden="${isHidden}"
              >${data[currentCol]}</td>

Because I cannot  break into a new line at ${data[currentCol]} in order to prevent empty spaces in <td> element which causes the element to display with white space. Therefore, I have to disabled eslint rule there.